### PR TITLE
plan: make show no longer a logical/physical plan

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

File: plan/logical_plan_test.go [hunk 1]
```diff
@@ -628,7 +628,7 @@ func (s *testPlanSuite) TestPlanBuilder(c *C) {
 		},
 		{
 			sql:  "show columns from t where `Key` = 'pri' like 't*'",
-			plan: "*plan.Show->Sel([eq(cast(key), 0)])",
```
Comment (XuHuaiyu) [inline plan/logical_plan_test.go]: should we add the condition in explain info for show 
like show(selections)

**Comment by coocood** [comment]
LGTM

File: plan/stringer.go [hunk 2]
```diff
@@ -119,6 +119,8 @@ func toString(in Plan, strs []string, idxs []int) ([]string, []int) {
 		str = "Lock"
 	case *ShowDDL:
 		str = "ShowDDL"
+	case *Show:
+		str = fmt.Sprintf("Show(%s)", x.Conditions)
```
Comment (XuHuaiyu) [inline plan/stringer.go]: check `if len(x.Conditions)  == 0` and `len(x.Conditions)  != 0`

File: executor/builder.go [hunk 3]
```diff
@@ -287,7 +287,14 @@ func (b *executorBuilder) buildShow(v *plan.Show) Executor {
...
 		e.User = e.ctx.GetSessionVars().User
 	}
-	return e
+	if len(v.Conditions) == 0 {
+		return e
+	}
+	sel := &SelectionExec{
+		baseExecutor: newBaseExecutor(v.Schema(), b.ctx, e),
+		filters:      v.Conditions,
+	}
```
Comment (zz-jason) [inline executor/builder.go]:
```go
sel.supportChunk = true
```

**Pull request merged**